src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1113,6 +1095,32 @@ Reader.SingleResult ToResult(TimeSeriesOperation.AppendOperation element)
             return context.LastDatabaseChangeVector;
         }
 
+        private bool ValueTooFar(TimeSeriesSegmentHolder segmentHolder, Reader.SingleResult current)
+        {
+            var deltaInMs = (current.Timestamp.Ticks / 10_000) - segmentHolder.BaselineMilliseconds;
+            return deltaInMs >= int.MaxValue;
+        }
+
+        private static bool EnsureNumberOfValues(DocumentsOperationContext context, int segmentNumberOfValues, Reader.SingleResult current)
+        {
+            if (segmentNumberOfValues > current.Values.Length)
+            {
+                using (context.Allocator.Allocate(segmentNumberOfValues * sizeof(double), out ByteString fullValuesBuffer))
+                {
+                    var fullValues = new Span<double>(fullValuesBuffer.Ptr, segmentNumberOfValues);
+                    current.Values.Span.CopyTo(fullValues);
+                    for (int i = current.Values.Length; i < fullValues.Length; i++)
+                    {
+                        fullValues[i] = double.NaN;
+                    }
+
+                    current.Values = new Memory<double>(fullValues.ToArray());
+                }
```

Comment:
You are returning the allocation here, but still using it.
This is _not_ valid

Reply:
`fullValues.ToArray()` copy them to managed memory?

Reply:
Oh, correct, but then why bother will all the song and ance? Just allocate them once directly and save all the complexity.

Reply:
right.